src/std/iter.ss

```diff
@@ -129,7 +129,9 @@ package: std
             (set! (cdr e) (##fx- limit 1))
             value)
           iter-end))))
-  (make-iterator (cons start count) next))
+  (if (andmap integer? [start count step])
```

Comment:
actually only the count needs to be an integer, the start and step can be any number.

Reply:
I see. In this case I would expect iteration to stop at 9.85:

```scheme
> (import :std/iter)                            
> (for (x (in-range 10 0.25 1.2)) (displayln x))
.25
1.45
2.65
3.8499999999999996
5.05
6.25
7.45
8.65
9.85
11.049999999999999
```

Or is that intentional? I would actually rather have `in-range` work on integers and maybe `in-rangef` work on floats. What do you think?

Reply:
that's an artifact of floating point numbers, it's unavoidable.
I think of it as the `iota` of iterators, and if we can do it with a single construct why use a second one?